### ISSUE TITLE
Building HOL4 will crash with latest Apple update of Xcode

### ISSUE THREAD

**Issue by jhlchan** [HOL-Theorem-Prover/HOL#1147]
This issue affects Apple users of HOL4. Apple recently released Command Line Tools for Xcode 15 (September 18, 2023). The new linker is incompatible with building HOL4:

```
poly < tools/smart-configure.sml
Poly/ML 5.9 Release
...
Making mllex
ld: LINKEDIT overlap of start of LINKEDIT and symbol table in '/private/var/folders/k5/_29n92td6yqcvzm2pf69n1n40000gn/T/polyobj.8335.o' in '/private/var/folders/k5/_29n92td6yqcvzm2pf69n1n40000gn/T/polyobj.8335.o'
clang: error: linker command failed with exit code 1 (use -v to see invocation)
Failed to make mllex (Fail "Command failed")
```

As the tools of HOL4 are deleted to be rebuild during configuration, this results in the loss of all tools, including Holmake and mkmunge. Current temporary solution to remedy the situation is:

1. Go to Apple Developer, More Downloads\
[https://developer.apple.com/download/all/?q=xcode](https://developer.apple.com/download/all/?q=xcode)
2. Download and install Command Line Tools for Xcode 14.3.1 (June 1, 2023).

After reverting from Xcode 15 to Xcode 14.3.1, building HOL4 will proceed as usual.

Some hints for fixing this bug:

Xcode 15 Release Notes\
[https://developer.apple.com/documentation/xcode-release-notes/xcode-15-release-notes](https://developer.apple.com/documentation/xcode-release-notes/xcode-15-release-notes)
```
Linking: New Features
A new linker has been written to significantly speed up static linking. It’s the default for all macOS, iOS, tvOS and visionOS binaries and anyone using the “Mergeable Libraries” feature.
The classic linker can still be explicitly requested using -ld64, and will be removed in a future release.  (108915312)
```

**Comment by mn200** [comment]
Thanks for this documentation!

**Comment by viljamivirolainen** [comment]
Just in case somebody else also encounters these issues.

I was not able to install the file from https://developer.apple.com/download/all/?q=xcode, there was error of too new OS (macOS Sonoma 14.0) so I used xcodes to install the specific version:
https://github.com/XcodesOrg/xcodes

Then there was an issue with MLton, so I created poly-includes.ML and added  "val MLTON = NONE;". 

Finally there was an issue of:
`ld: can't open output file for writing: /Users/viljamivirolainen/HOL/bin/Holmake, errno=2 for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
Failed to make Holmake (Fail "Command failed")`

But that was solved by "mkdir bin" in HOL directory.
